beacon_node/execution_layer/src/lib.rs

```diff
@@ -260,6 +354,40 @@ pub struct ExecutionLayer<T: EthSpec> {
     inner: Arc<Inner<T>>,
 }
 
+/// This function will return the percentage difference between 2 U256 values, using `base_value`
+/// as the denominator. It is accurate to 7 decimal places which is about the precision of
+/// an f32.
+///
+/// If some error is encountered in the calculation, None will be returned.
+fn percentage_difference_u256(base_value: Uint256, comparison_value: Uint256) -> Option<f32> {
+    if base_value == Uint256::zero() {
+        return None;
+    }
+    // this is the total supply of ETH in WEI
+    let max_value = Uint256::from(12u8) * Uint256::exp10(25);
+    if base_value > max_value || comparison_value > max_value {
+        return None;
+    }
+
```

Comment:
I guess this is really just a sanity check on the value, as a valid payload is unlikely to hit this value. Is it really necessary to do this check though? (btw this would be 125 million xDai on Gnosis Chain)

Reply:
The check is integral to the following assumption though: 
```
// Now we should be able to calculate the difference without division by zero or overflow
```

So i think we should leave it for now and we could potentially re-work the rest of the method in a different PR to handle division by zero/overflow otherwise